lib/segment/src/index/field_index/mmap_point_to_values.rs

```diff
@@ -0,0 +1,385 @@
+        Self::read_from_prefix(bytes)
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8]) -> OperationResult<()> {
+        self.write_to_prefix(bytes)
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+}
+
+impl<'a> MmapValue<'a> for FloatPayloadType {
+    fn mmaped_size(&self) -> usize {
+        std::mem::size_of::<Self>()
+    }
+
+    fn read_from_mmap(bytes: &[u8]) -> Option<Self> {
+        Self::read_from_prefix(bytes)
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8]) -> OperationResult<()> {
+        self.write_to_prefix(bytes)
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+}
+
+impl<'a> MmapValue<'a> for GeoPoint {
+    fn mmaped_size(&self) -> usize {
+        2 * std::mem::size_of::<f64>()
+    }
+
+    fn read_from_mmap(bytes: &[u8]) -> Option<Self> {
+        Some(Self {
+            lon: f64::read_from_prefix(bytes)?,
+            lat: bytes
+                .get(std::mem::size_of::<f64>()..)
+                .and_then(f64::read_from_prefix)?,
+        })
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8]) -> OperationResult<()> {
+        self.lon
+            .write_to_prefix(bytes)
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+        bytes
+            .get_mut(std::mem::size_of::<f64>()..)
+            .and_then(|bytes| self.lat.write_to_prefix(bytes))
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))
+    }
+}
+
+impl<'a> MmapValue<'a> for &'a str {
+    fn mmaped_size(&self) -> usize {
+        self.as_bytes().len() + std::mem::size_of::<u32>()
+    }
+
+    fn read_from_mmap(bytes: &'a [u8]) -> Option<&'a str> {
+        let size = u32::read_from_prefix(bytes)? as usize;
+
+        let bytes = bytes.get(std::mem::size_of::<u32>()..std::mem::size_of::<u32>() + size)?;
+        std::str::from_utf8(bytes).ok()
+    }
+
+    fn write_to_mmap(&self, bytes: &mut [u8]) -> OperationResult<()> {
+        u32::write_to_prefix(&(self.len() as u32), bytes)
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+
+        let bytes = bytes
+            .get_mut(std::mem::size_of::<u32>()..std::mem::size_of::<u32>() + self.len())
+            .ok_or_else(|| OperationError::service_error(NOT_ENOUGHT_BYTES_ERROR_MESSAGE))?;
+
+        self.as_bytes()
+            .iter()
+            .enumerate()
+            .for_each(|(i, &b)| bytes[i] = b);
+        Ok(())
+    }
+}
+
+/// Flattened memmaped points-to-values map
+/// It's an analogue of `Vec<Vec<N>>` but in memmaped file.
+/// This structure doesn't support adding new values, only removing.
+/// It's used in mmap field indices like `MmapMapIndex`, `MmapNumericIndex`, etc to store points-to-values map.
+/// This structure is not generic to avoid boxing lifetimes for `&str` values.
+pub struct MmapPointToValues {
```

Comment:
Please move to a separate file

Reply:
WDYM? It's a separate file `mmap_point_to_values.rs` for `MmapPointToValues` structure. Another definitions are traits related to this structure